app/sidecar_query_server.go

```diff
@@ -155,7 +155,27 @@ func NewSideCarQueryServer(appCodec codec.Codec, config domain.Config, logger lo
 		quotePriceUpdateWorker.RegisterListener(chainInfoUseCase)
 
 		// Initialize ingest handler and usecase
-		ingestUseCase, err := ingestusecase.NewIngestUsecase(poolsUseCase, routerUsecase, chainInfoUseCase, appCodec, quotePriceUpdateWorker, logger)
+		ingestUseCase, err := ingestusecase.NewIngestUsecase(
+			poolsUseCase,
+			routerUsecase,
+			chainInfoUseCase,
+			appCodec,
+			quotePriceUpdateWorker,
+			[]func(height uint64){
+				func(height uint64) {
+					// do not block, run as a go routine
+					tokensUsecase.UpdateAssetsAtHeightInterval(
+						height,
+						200, // TODO config
+						tokensUsecase.LoadTokensFromChainRegistry,
+						tokensUseCase,
+						config.ChainRegistryAssetsFileURL,
+					)
+					logger.Info("callback block processing", zap.Uint64("height", height))
+				},
+			},
```

Comment:
Not super sure why we need this wiring.

Can we instead have ingest handler call this:
```
tokensUsecase.UpdateAssetsAtHeightInterval(height)
```

For all the config parameters, we can set them on tokens use case in constructor

Reply:
Original idea I had was to have more generic callback mechanism, where `ingester` just calls any number of `callback` functions with signature of `func(height uint64)`, and in turn specifics of constructing actual wiring were moved to the program entry point. 

As per your suggestion I have updated to just call `tokensUsecase.UpdateAssetsAtHeightInterval(height)` in the ingest handler.

Reply:
Original idea I had was to have more generic callback mechanism, where `ingester` just calls any number of `callback` functions with signature of `func(height uint64)`, and in turn specifics of constructing actual wiring were moved to the program entry point. 

As per your suggestion I have updated to just call `tokensUsecase.UpdateAssetsAtHeightInterval(height)` in the ingest handler.